temporalio/contrib/openai_agents/_model_parameters.py

```diff
@@ -41,7 +43,19 @@ class ModelActivityParameters:
     versioning_intent: Optional[VersioningIntent] = None
     """Versioning intent for the activity."""
 
-    summary_override: Optional[str] = None
+    summary_override: Optional[
+        Union[
+            str,
+            Callable[
+                [
+                    Optional[Agent[Any]],
+                    Optional[str],
+                    Union[str, list[TResponseInputItem]],
+                ],
+                str,
+            ],
```

Comment:
This is complicated enough to deserve a `class ModelSummaryProvider(ABC):` + `@abstractmethod def provide` type of thing IMO. It's hard to give names to these params with this complex of a callable. But up to you.